### PR TITLE
FEATURE/ElvaRivle/BaseModal

### PR DESCRIPTION
**Pull request by ElvaRivle**
ElvaRivle/Feature/BaseModal (DON'T MERGE) 

--------------------------------------------------------------------
 MAIN
 * refactored folder structure and data flow 
 * resolved PR comments
--------------------------------------------------------------------
 OTHER
 * /
--------------------------------------------------------------------
 TODO
 * TODO in AddEmployeeModal.js

### CODE REVIEW

**Comment by ElvaRivle** [comment]
FEATURE/ElvaRivle/BaseModal

--------------------------------------------------------------------
 MAIN
 * modalItem component added
 * populated modal from props
 * basic styling of modal
--------------------------------------------------------------------
 OTHER
 * modal example in Dashboard
--------------------------------------------------------------------
 TODO
 * FIXME's left to fix

File: src/components/dashboard/Dashboard.js [hunk 1]
```diff
@@ -1,10 +1,43 @@
...
+      ],
+      id: "department",
+      name: "department",
+      labelText: "Department: ",
+    },
+  ]
+
   return (
     <>
+      <button onClick={() => setIsOpen(true)}>
```
Comment (becarselman) [inline src/components/dashboard/Dashboard.js]: move this onClick function outside of jsx. In this case function will be defined on each render.

File: src/components/dashboard/Dashboard.js [hunk 2]
```diff
@@ -1,10 +1,43 @@
...
+        Click to Open Modal
+      </button>
+
+      <Modal handleClose={() => setIsOpen(false)} isOpen={isOpen} items = { employees } />
```
Comment (becarselman) [inline src/components/dashboard/Dashboard.js]: same for handleClose

File: src/components/modal/modal.scss [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    overflow: hidden;
+    z-index: 999;
+    padding: 40px 20px 20px;
+}
+
+.modal-content {
+    //width: 70%;
+    //height: 70%;
+    background-color: #E6F2ED;
+    color: #000;
```
Comment (becarselman) [inline src/components/modal/modal.scss]: move colors in variables file for colors. and if already exist reuse it

File: src/components/modal/modal.scss [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    //font-size: 2rem;
```
Comment (becarselman) [inline src/components/modal/modal.scss]: move commented code

File: src/components/modal/modalItem.js [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import "./modalItem.scss"
+
+const ModalItem = (props) => {
```
Comment (becarselman) [inline src/components/modal/modalItem.js]: you can use useMemo for this type of component

**Review by becarselman — CHANGES_REQUESTED** [review]

**Comment by ElvaRivle** [comment]
FEATURE/ElvaRivle/BaseModal

--------------------------------------------------------------------
 MAIN
 * fixed issues from PR review
--------------------------------------------------------------------
 OTHER
 * created [wiki](https://github.com/becarselman/ant-colony-pharaoh-frontend/wiki/Modal-Usage) for usage of modal
--------------------------------------------------------------------
 TODO
 * modal styling

File: src/components/_addEmployeeModal/AddEmployeeModal.js [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+      surname,
+      department,
+      salary,
+      stack
+    })
+  }
+
+  return (
+    <>
+      <Modal handleClose={() => {
```
Comment (becarselman) [inline src/components/_addEmployeeModal/AddEmployeeModal.js]: you can create separate function handleClose and just pass here. avoid using inline function in jsx

File: src/components/modal/Modal.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+    )
+  })
+
+  return (
+    <ReactPortal wrapperId={"modal-wrapper"}>
+      <div className="modal">
+        <CloseOutlined onClick={handleClose} className="close-btn"/>
+        <div className="modal-content">
+          { mappedItems }
+          {
```
Comment (becarselman) [inline src/components/modal/Modal.js]: move this outside of jsx. It will be more readable

File: src/components/modal/ModalItem.js [hunk 8]
```diff
@@ -0,0 +1,60 @@
+import "./modalItem.scss"
+import Select from "react-select";
+
+const ModalItem = (props) => {
```
Comment (becarselman) [inline src/components/modal/ModalItem.js]: this is good but my personal preference is to destruct items from props and than I don't need to type props each time. But this is also ok

File: src/components/modal/modalItem.scss [hunk 9]
```diff
@@ -0,0 +1,16 @@
+label {
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-start;
+  //line-height: 26px;
```
Comment (becarselman) [inline src/components/modal/modalItem.scss]: remove commented code if you don't need

**Review by becarselman — COMMENTED** [review]
This looks good just fix this small comments. Also I hope you are tested everything create/ edit

File: src/components/modal/ModalItem.js (hunk 8 above)
Comment (ElvaRivle) [reply src/components/modal/ModalItem.js]: this component shall be deleted when refactoring starts, but I'll keep that in mind

**Review by becarselman — APPROVED** [review]

**Review by TarikVelic — APPROVED** [review]

**Pull request merged**